packages/wrangler/src/__tests__/r2/helpers.test.ts

```diff
@@ -0,0 +1,25 @@
+import { eventNotificationHeaders } from "../../r2/helpers";
+import type { ApiCredentials } from "../../user";
+
+describe("event notifications", () => {
+	const credentials: ApiCredentials[] = [
+		{
+			authEmail: "test@example.com",
+			authKey: "some-big-secret",
+		},
+		{ apiToken: "some-api-token" },
+	];
+	test.each(credentials)("eventNotificationHeaders", (creds) => {
+		const result = eventNotificationHeaders(creds);
+		if ("authEmail" in creds) {
+			expect(result).toMatchObject({
+				"X-Auth-Key": creds.authKey,
+				"X-Auth-Email": creds.authEmail,
+			});
+		} else {
+			expect(result).toMatchObject({
+				Authorization: `Bearer ${creds.apiToken}`,
+			});
+		}
+	});
+});
```

Comment:
NIT given that there are only two items in `credentials` and you end up having to do an `if` statement in the test there is little point in using `test.each` here:

```suggestion
describe("event notifications", () => {
  test("auth email eventNotificationHeaders", () => {
    const result = eventNotificationHeaders({
			authEmail: "test@example.com",
			authKey: "some-big-secret",
		});
			expect(result).toMatchObject({
				"X-Auth-Key": creds.authKey,
				"X-Auth-Email": creds.authEmail,
			});
  });
  
  test("API token eventNotificationHeaders", () => {
		const result = eventNotificationHeaders({ apiToken: "some-api-token" });
		expect(result).toMatchObject({
			Authorization: `Bearer ${creds.apiToken}`,
		});
	});
});
```